source/Missions/Services/Arena/CoopArenaController.cs

```diff
@@ -89,52 +111,72 @@ public Agent AddPlayerToArena()
             }
 
             // get a random spawn point
-            MatrixFrame randomElement = spawnFrames.GetRandomElement();
+            Random rand = new Random();
+            MatrixFrame randomElement = spawnFrames[rand.Next(spawnFrames.Count)];  
 
 
             // spawn an instance of the player (controlled by default)
-            return SpawnPlayerAgent(CharacterObject.PlayerCharacter, randomElement);
+            SpawnPlayerAgent(CharacterObject.PlayerCharacter, randomElement);
+
+            Agent.Main.SetTeam(Mission.Current.PlayerTeam, false);
+
+            IEnumerable<CharacterObject> listC = CharacterObject.All.Where(x => !x.IsHero);
+            Random r = new Random();
+
+            _tempAi = SpawnAgent(randomElement.origin, listC.ElementAt(r.Next(listC.Count())), false);
         }
 
-        // Spawn an agent based on its character object and frame. For now, Main agent character object is used
-        // This should be the real character object in the future
+        /// <summary>

```

Comment:
Move this after [line 133](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/pull/494/files#diff-45dc98f684924d6f81367819b5088ff691c595630067fc2442386cfb8585c1cfR133) so when you hover over any usage of SpawnPlayerAgent the summary appears